### PR TITLE
inductor: convert view to reshape before doing fake_tensor_prop at freezing step

### PR DESCRIPTION
**Pull request by XiaobingSuper**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #104614
* __->__ #104612

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @ngimel @yf225 @chenyang78 @kadeng

### CODE REVIEW

File: torch/_inductor/freezing.py [hunk 1]
```diff
@@ -222,7 +222,12 @@ def freeze(
     aot_autograd_gm.recompile()
 
     from torch._inductor.compile_fx import fake_tensor_prop
+    from torch._inductor.fx_passes.post_grad import view_to_reshape
 
+    # We have convert conv's weight to channels last which may meet error for .view
+    # when doing fake_tensor_prop. So we need to convert view to reshape first.
+    # see https://github.com/pytorch/pytorch/blob/main/torch/_inductor/compile_fx.py#L472.
```
Comment (jgong5) [inline torch/_inductor/freezing.py:229]: Seems not good to place a code link without commit id here (things might change in the future and the code lines don't match). How about add a "note anchor" (like `[NOTE: view to reshape conversion]`) in compile_fx.py#472 and refer to the note anchor here?

**Review by jgong5 — APPROVED** [review]
LGTM. A minor suggestion on the comment.

File: torch/_inductor/freezing.py (hunk 1 above)
Comment (XiaobingSuper) [reply torch/_inductor/freezing.py]: updated.

File: torch/_inductor/freezing.py [hunk 2]
```diff
@@ -222,7 +222,12 @@ def freeze(
     aot_autograd_gm.recompile()
 
     from torch._inductor.compile_fx import fake_tensor_prop
+    from torch._inductor.fx_passes.post_grad import view_to_reshape
```
Comment (eellison) [inline torch/_inductor/freezing.py:225]: this can be imported at the top of the file

**Review by eellison — APPROVED** [review]

**Review by shunting314 — APPROVED** [review]

File: torch/_inductor/freezing.py (hunk 2 above)
Comment (XiaobingSuper) [reply torch/_inductor/freezing.py]: changed.

**Pull request closed without merging**